packages/langium-sql/src/sql-type-computation.ts

```diff
@@ -158,6 +159,9 @@ export class SqlTypeComputer implements TypeComputer {
         if(isIdentifierAsStringLiteral(node)) {
             return Types.Char();
         }
+        if(isCaseExpression(node)) {
+            return undefined;
```

Comment:
Case expressions HAVE a return type. You might need a helper function to get the common type of all WHEN/ELSE-results or look if all results are of same type.